### PR TITLE
GIX-1823: Set sns aggregator store flag true for tests

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

Set ENABLE_SNS_AGGREGATOR_STORE `true` for tests.

# Changes

* Set ENABLE_SNS_AGGREGATOR_STORE to `true` in jest.setup.ts
* Use snsQueryStore in sns services spec file. This should have never been setSnsProjects.
* Add more parameters to `aggregatorSnsMockWith`.
* Use aggregatorStore in setSnsProjects test util.

# Tests

Only test changes.

# Todos

- [x] Add entry to changelog (if necessary).

### CODE REVIEW

**Comment by lmuntaner** [comment]
@dskloetd please review

File: frontend/src/tests/lib/routes/NeuronDetail.spec.ts [hunk 1]
```diff
@@ -123,10 +123,8 @@ describe("NeuronDetail", () => {
       // Load SNS projects after rendering to make sure we don't load
       // NnsNeuronDetail instead, which was a bug we had.
       await loadSnsProjects();
-      expect(await po.isContentLoaded()).toBe(false);
-      await waitFor(async () => {
-        expect(await po.isContentLoaded()).toBe(true);
-      });
+      expect(await po.isContentLoaded()).toBe(true);
```
Comment (lmuntaner) [inline frontend/src/tests/lib/routes/NeuronDetail.spec.ts:126]: Loading the project was faster and the first expect was already failing.

There is a previous check that content is not loaded. Therefore, it's covered.

File: frontend/jest-setup.ts [hunk 2]
```diff
@@ -36,7 +36,7 @@ jest.mock("./src/lib/utils/env-vars.utils.ts", () => ({
       ENABLE_CKTESTBTC: true,
       ENABLE_ICP_ICRC: false,
       ENABLE_INSTANT_UNLOCK: true,
-      ENABLE_SNS_AGGREGATOR_STORE: false,
+      ENABLE_SNS_AGGREGATOR_STORE: true,
```
Comment (dskloetd) [inline frontend/jest-setup.ts:39]: I'm not really comfortable with this.
I'd rather keep the test coverage of the disabled flag until the flag is actually enabled.
But any tests that have different behavior with or without the flag should ideally have a twin where the flag is enabled explicitly.

File: frontend/src/tests/lib/pages/SnsProposals.spec.ts [hunk 3]
```diff
@@ -49,21 +52,21 @@ describe("SnsProposals", () => {
...
     ]);
     // Reset to default value
     page.mock({ data: { universe: rootCanisterId.toText() } });
   });
 
   describe("logged in user", () => {
-    const functionName = "test_function";
-    const functionId = BigInt(3);
     beforeEach(() => {
-      fakeSnsGovernanceApi.addNervousSystemFunctionWith({
```
Comment (dskloetd) [inline frontend/src/tests/lib/pages/SnsProposals.spec.ts:62]: Was this change necessary?
Don't we still have a separate store that is loaded this way?

File: frontend/src/tests/lib/services/sns.services.spec.ts [hunk 4]
```diff
@@ -124,16 +126,17 @@ describe("sns-services", () => {
           direct_participant_count: [],
           cf_neuron_count: [],
         };
-        setSnsProjects([
```
Comment (dskloetd) [inline frontend/src/tests/lib/services/sns.services.spec.ts:127]: How about making the behavior of `setSnsProjects` depend on the flag?

Otherwise at least keep the existing function with a different name. Reverting back to `snsQueryStore.setData(responses);` now seems silly.

File: frontend/jest-setup.ts (hunk 2 above)
Comment (lmuntaner) [reply frontend/jest-setup.ts:39]: I can also change it in dfx.json except for mainnet. That was my next plan.

File: frontend/src/tests/lib/pages/SnsProposals.spec.ts (hunk 3 above)
Comment (lmuntaner) [reply frontend/src/tests/lib/pages/SnsProposals.spec.ts:62]: The functions are not loaded if they already exist in the snsAggregatorStore.

Therefore, the governance api is not called because we set the projects before.

File: frontend/src/tests/lib/services/sns.services.spec.ts (hunk 4 above)
Comment (lmuntaner) [reply frontend/src/tests/lib/services/sns.services.spec.ts:127]: This is not reverting back, this should not have been setSnsProjects. This tests literally deal with snsQueryStore. Not with SNSes or snsSummaries.

Comment (lmuntaner) [reply frontend/src/tests/lib/services/sns.services.spec.ts:127]: I can change this in another PR if you prefer.

File: frontend/jest-setup.ts (hunk 2 above)
Comment (lmuntaner) [reply frontend/jest-setup.ts:39]: I think we should change the test before changing mainnet. I'm more comfortable with that.

File: frontend/src/tests/lib/pages/SnsProposals.spec.ts (hunk 3 above)
Comment (dskloetd) [reply frontend/src/tests/lib/pages/SnsProposals.spec.ts:62]: Can you keep the old tests and disable the feature flag in it to make sure it continues to work until we have actually enabled the feature flag in mainnet?

File: frontend/src/tests/lib/services/sns.services.spec.ts (hunk 4 above)
Comment (dskloetd) [reply frontend/src/tests/lib/services/sns.services.spec.ts:127]: It *is* reverting back. You're saying that reverting back is intentional. But the other function was more readable.
And the whole point of that function was that you don't have to change individual calls to it.

File: frontend/jest-setup.ts (hunk 2 above)
Comment (dskloetd) [reply frontend/jest-setup.ts:39]: When do you plan to set the flag on mainnet?
At least make sure that every test that had to change has a twin where the flag is still disabled.

Comment (lmuntaner) [reply frontend/jest-setup.ts:39]: Only the sns.services needed it and they are added here.

The other functionality uses snsSummariesStore, which has also tests with the feature flag enabled and disabled.

Comment (lmuntaner) [reply frontend/jest-setup.ts:39]: I want to try the next release again, and I believe that next week we could turn it on.

File: frontend/src/tests/lib/pages/SnsProposals.spec.ts (hunk 3 above)
Comment (lmuntaner) [reply frontend/src/tests/lib/pages/SnsProposals.spec.ts:62]: Yes, sounds good.

File: frontend/src/tests/lib/services/sns.services.spec.ts (hunk 4 above)
Comment (lmuntaner) [reply frontend/src/tests/lib/services/sns.services.spec.ts:127]: This tests deal direclty with snsQueryStore, not with setting SNSes. Therefore, it makes sense that here it specifically sets snsQueryStore.

I won't have to change individual calls to it after changing the flag, because this doesn't deal with the snsAggregatorStore. The only changes regardless of the feature flag are about setting or not setting the snsQueryStore (directly). But the snsAggregatorStore doesn't play any role in this tests.

**Comment by lmuntaner** [comment]
@dskloetd ready for another review

**Review by dskloetd — APPROVED** [review]
